File: yologram-api-v2/app/domain/comment/post_query_client.py
from typing import Protocol

from sqlalchemy.orm import Session

from app.domain.pms.repository import PostRepository


class PostQueryClient(Protocol):
    """
    comment → pms 도메인 경계 호출 추상화 (대상 글 존재 검증).
    모놀리식에서는 pms 리포지토리를 직접 호출(LocalPostQueryClient),
    MSA 분리 시 pms-api HTTP 호출 구현으로 교체한다.
    """

    def exists(self, post_id: int) -> bool:
        """post_id의 게시글이 존재하면 True."""
        ...


class LocalPostQueryClient:

    def __init__(self, db: Session):
        self.repository = PostRepository(db)

    def exists(self, post_id: int) -> bool:
        return self.repository.find_by_id(post_id) is not None

File: yologram-api-v1/src/main/kotlin/link/yologram/api/v1/domain/comment/service/PostQueryClient.kt
package link.yologram.api.v1.domain.comment.service

/**
 * comment → pms 도메인 경계 호출 추상화 (대상 게시글 존재 검증).
 * 모놀리식에서는 pms 리포지토리를 직접 호출(LocalPostQueryClient),
 * MSA 분리 시 post-api HTTP 호출 구현으로 교체한다.
 */
interface PostQueryClient {
    /** postId 게시글이 존재하면 true. */
    fun exists(postId: Long): Boolean
}
